### DISCUSSION TITLE
What's the difference between `globalTypes` and `parameters`?

### DISCUSSION THREAD

**Discussion by bytrangle** [storybookjs/storybook#18446] [Help] [answered]
The `.storybook/preview.js` file can export both `globalTypes` and `parameters`.

This [example](https://storybook.js.org/docs/react/writing-stories/parameters#global-parameters) shows that one can configure background color for all stories using the `parameters` export.

But this [example for using `globalTypes` to set theme](https://storybook.js.org/docs/react/essentials/toolbars-and-globals#global-types-and-the-toolbar-annotation) is quite similar.

Looking at other examples, most of them tend to use `parameters` in a more declarative static nature, such as in the `angular-cli` example:

```
export const parameters = {
  docs: {
    inlineStories: true,
  },
  options: {
    storySort: {
      order: ['Welcome', 'Core ', 'Addons ', 'Basics '],
    },
  },
};
```

So when should we use which?

**Comment by shilman** [comment, marked as answer]
- [Parameters](https://storybook.js.org/docs/react/writing-stories/parameters) never change, so they are good for specifying options for addons (e.g. the URL of a design resource for a story)
- [Args](https://storybook.js.org/docs/react/writing-stories/args) are dynamic values that get passed into stories and can be manipulated by UI controls in the addon panel (e.g. the label of a button). When the args for a story changes, the story re-renders.
- [Globals](https://storybook.js.org/docs/react/essentials/toolbars-and-globals#globals) are dynamic values that are set globally across Storybook and can be manipulated by UI controls in the toolbar (e.g. the theme, i18n locale, etc.). When the global value changes, all the stories on the current page re-render.

**Reply by bytrangle** [reply]
Thank you for the splendid answer.
